node/network/dispute-distribution/src/sender/mod.rs

```diff
@@ -190,8 +190,24 @@ impl DisputeSender {
 		dispute: (SessionIndex, CandidateHash),
 	) -> Result<()> {
 		let (session_index, candidate_hash) = dispute;
-		// We need some relay chain head for context for receiving session info information:
-		let ref_head = self.active_sessions.values().next().ok_or(NonFatal::NoActiveHeads)?;
+		// A relay chain head is required as context for receiving session info information from runtime and
+		// storage. We will iterate `active_sessions` to find a suitable head. We assume that there is at
+		// least one active head which, by `session_index`, is at least as recent as the `dispute` passed in.
+		// We need to avoid picking an older one from a session that might not yet exist in storage.
+		// Related to https://github.com/paritytech/polkadot/issues/4730.
```

Comment:
nit: always keep trailing space after links, or the `.` will become part of the link, or even better use `<link>`
```suggestion
		// Related to <https://github.com/paritytech/polkadot/issues/4730> .
```